rclcpp/src/rclcpp/time_source.cpp

```diff
@@ -105,17 +105,36 @@ void TimeSource::attachNode(
       rclcpp::to_string(use_sim_time_param.get_type()).c_str());
   }
 
-  // TODO(tfoote) use parameters interface not subscribe to events via topic ticketed #609
-  parameter_subscription_ = rclcpp::AsyncParametersClient::on_parameter_event(
```

Comment:
This shouldn't be deleted.
`set_on_parameters_set_callback` callbacks are for rejecting/accepting parameter, but not for updating other things based on the result (multiple callbacks can be registered, and you don't know if the parameter will finally be accepted or not).
For that we're using `on_parameter_event`, though it may be good to have a "local" alternative to it (the current approach is creating a subscription).